File: specification/specs/x_pack/slm/SnapshotLifecyclePolicyMetadata.ts
class SnapshotLifecyclePolicyMetadata {
  in_progress: SnapshotLifecycleInProgress
  last_failure: SnapshotLifecycleInvocationRecord
  last_success: SnapshotLifecycleInvocationRecord
  /** @prop_serializer DateTimeOffsetEpochMillisecondsFormatter */
  modified_date_millis: Date
  /** @prop_serializer DateTimeOffsetEpochMillisecondsFormatter */
  next_execution_millis: Date
  policy: SnapshotLifecyclePolicy
  version: integer
}

File: output/csharp/XPack/Slm/SnapshotLifecyclePolicyMetadata.cs
using Nest.XPack;
using Nest.Internal;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.XPack {

	public class SnapshotLifecyclePolicyMetadata  {
		
		[DataMember(Name="in_progress")]
		public SnapshotLifecycleInProgress InProgress { get; set; }


		[DataMember(Name="last_failure")]
		public SnapshotLifecycleInvocationRecord LastFailure { get; set; }


		[DataMember(Name="last_success")]
		public SnapshotLifecycleInvocationRecord LastSuccess { get; set; }


		[DataMember(Name="modified_date_millis")]
		public DateTimeOffset ModifiedDateMillis { get; set; }


		[DataMember(Name="next_execution_millis")]
		public DateTimeOffset NextExecutionMillis { get; set; }


		[DataMember(Name="policy")]
		public SnapshotLifecyclePolicy Policy { get; set; }


		[DataMember(Name="version")]
		public int Version { get; set; }

	}
}
